nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm

```diff
@@ -159,3 +255,139 @@
 
 /obj/machinery/ship_weapon/energy/beam/admin //ez weapon for quickly testing.
 	charge_per_shot = 0
+
+
+
+/obj/machinery/ship_weapon/energy/proc/process_heat()//heat management. don't push your weapons too hard. actual heat generation is in _ship_weapons.dm
+
+	for(var/obj/machinery/cooling/cooler/C in coolers)
+		if(!(C.machine_stat & (BROKEN|NOPOWER|MAINT)))
+			cooling_amount++
+	for(var/obj/machinery/cooling/storage/C in storages)
+		if(!(C.machine_stat & (BROKEN|NOPOWER|MAINT)))
+			storage_amount++
+	max_heat = initial(max_heat) + (storage_amount*storage_rate)
+	var/H = heat-cooling_amount*heat_rate
+	if(heat > 0)
+		heat = max((H),0)
+	if((weapon_state == STATE_OVERLOAD) & (heat <= (max_heat/50)))
+		weapon_state = STATE_NOTHING
+	if(weapon_state == STATE_OVERLOAD)
+		return
+	if(weapon_state == STATE_VENTING)
+		if(heat <= max_heat-ventnumber)
+			weapon_state = STATE_NOTHING
+			return
+		heat = max(((H*1.2)-H),0)
+		return
+	if(heat >= max_heat)
+		overload()
+
+
+/obj/machinery/ship_weapon/energy/proc/overload() //this is what happens when you can't control yourself
+	playsound(src, malfunction_sound, 100, 1)
+	playsound(src, overheat_sound, 100, 1)
+	do_sparks(4, FALSE, src)
+	weapon_state = STATE_OVERLOAD
+	alignment = 0
+	freq = 0
+	say("WARNING! Critical heat density, emergency venting and shutdown initiated!")
+	atmos_spawn_air("water_vapor=200;TEMP=1000")
+	heat = max_heat
+	charge = 0
+	return
+
+/obj/machinery/ship_weapon/energy/proc/handle_alignment() //this is the basic bad stuff that happens, don't fire when your gun is at 0 alignment, or it'll blow itself up
+	var/turf/detonation_turf = get_turf(src)
+	if(heat >= (3*(max_heat/4)))
+		freq -= rand(1,4)
+	if(alignment <= 75)
+		if(prob(50))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+	if(alignment <= 50)
+		if(prob(45))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+			playsound(src, malfunction_sound, 100, 1)
+		if(prob(5))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 2, 3, flame_range = 2)
+	if(alignment <= 25)
+		if(prob(25))
+			do_sparks(4, FALSE, src)
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+		if(prob(25))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 3, 4, flame_range = 3)
+		if(prob(50))
+			var/list/shootat_turf = RANGE_TURFS(5,detonation_turf) - RANGE_TURFS(4, detonation_turf)
+			var/obj/item/projectile/beam/laser/P = new(detonation_turf)
+			//Shooting Code:
+			P.range = 6
+			P.preparePixelProjectile(pick(shootat_turf), detonation_turf)
+			P.fire()
+			freq -= rand(1,10)
+	alignment = max(alignment-(rand(0, 4)),0)
```

Comment:
```suggestion
	switch(alignment)
		if(51 to 75)
			if(prob(50))
				do_sparks(4, FALSE, src)
				freq -= rand(1,10)
		if(26 to 50)
			var/roll = roll(1,20)
			switch(roll)
				if(1 to 9)
					do_sparks(4, FALSE, src)
					freq -= rand(1,10)
					playsound(src, malfunction_sound, 100, 1)
				if(10)
					playsound(src, malfunction_sound, 100, 1)
					freq -= rand(1,10)
					explosion(detonation_turf, 0, 0, 2, 3, flame_range = 2)
		if(0 to 25)
			var/roll2 = roll(1,4)
			switch(roll2)
				if(1)
					do_sparks(4, FALSE, src)
					playsound(src, malfunction_sound, 100, 1)
					freq -= rand(1,10)
				if(2)
					playsound(src, malfunction_sound, 100, 1)
					freq -= rand(1,10)
					explosion(detonation_turf, 0, 0, 3, 4, flame_range = 3)
				if(3,4)
					var/list/shootat_turf = RANGE_TURFS(5,detonation_turf) - RANGE_TURFS(4, detonation_turf)
					var/obj/item/projectile/beam/laser/P = new(detonation_turf)
					//Shooting Code:
					P.range = 6
					P.preparePixelProjectile(pick(shootat_turf), detonation_turf)
					P.fire()
					freq -= rand(1,10)
	alignment = max(alignment-(rand(0, 4)),0)
```
The way you have written this originally, all of these conditions will fire at the same time if alignment is < 25, so use a switch statement instead. I also added switch statements to your switch statements, because I figured you didn't want all of your probs which suspiciously added up to 100% and did a bunch of the same stuff to proc at the same time either.